sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -300,7 +300,14 @@ def format_parameters(self, params):
                 p[0]: p[-1] for p in [p.partition("=") for p in query.split("&")]
             }
             params.update(existing_params)
-        query_params = ["{}={}".format(k, v) for k, v in params.items()]
+        query_params = []
+        for k, v in params.items():
+            if isinstance(v, str):
+                query_params.append("{}={}".format(k, v))
+            elif isinstance(v, list):
+                query_params.extend(["{}={}".format(k, w) for w in v])
+            else:
+                raise ValueError()
```

Comment:
Do we want to be that precise? Format could call automatically `__str__` on an object, with this code using integer will fail for example now.
What about:
```python
            if isinstance(v, list):
                query_params.extend(["{}={}".format(k, w) for w in v])
            else:
                query_params.append("{}={}".format(k, v))
```

Reply:
Good catch. My implementation is a potential breaking change. Thanks.